Add DeepX to export formats table and optimize arg docs (#24488)
ultralytics/ultralytics: docs/macros/export-args.md

| `imgsz`     | `int` or `tuple`  | `640`           | Desired image size for the model input. Can be an integer for square images (e.g., `640` for 640×640) or a tuple `(height, width)` for specific dimensions.                                                                                                                                                                                                |
| `keras`     | `bool`            | `False`         | Enables export to Keras format for [TensorFlow](https://www.ultralytics.com/glossary/tensorflow) SavedModel, providing compatibility with TensorFlow serving and APIs.                                                                                                                                                                                     |

> | `optimize`  | `bool`            | `False`         | Applies optimization for mobile devices when exporting to TorchScript, potentially reducing model size and improving [inference](https://docs.ultralytics.com/modes/predict) performance. Not compatible with NCNN format or CUDA devices. For DeepX, enables a higher compiler optimization which reduces inference latency and increases compilation time.                                                                                                                 |

Reviewer: 📝 **LOW**: This combines TorchScript- and DeepX-specific behavior into one description, but the compatibility note is still phrased as if it applies to `optimize` globally. As written, readers may infer that DeepX exports are also incompatible with CUDA devices. It would be clearer to scope the NCNN/CUDA restriction explicitly to TorchScript before introducing the DeepX behavior.

**Suggested change:**
```suggestion
| `optimize`  | `bool`            | `False`         | For TorchScript, applies optimization for mobile devices, potentially reducing model size and improving [inference](https://docs.ultralytics.com/modes/predict) performance; this is not compatible with NCNN format or CUDA devices. For DeepX, enables a higher compiler optimization which reduces inference latency and increases compilation time.                                                                                                                 |
```
